### DISCUSSION TITLE
What makes stable?

### DISCUSSION THREAD

**Discussion by mkopchenov** [termux/termux-packages#17817] [Q&A] [answered]
Can someone point me to the doc for when the package makes it into stable? 
I could not find it. 
Case in point: curl-8.3 and libcurl-8.3 were added to stable. 
There was no critical issue (latest was 8.2.1, ahead of many other distributions). 
So I am curious as to what are the rules.
Thanks.

**Comment by TomJo2000** [comment]
A "stable" release, generally is any new tagged release of a project.

Be that:
- a regular version bump, like `curl 8.2.1` to `8.3`
- a minor version bump, like `foo 0.7.5` to `0.7.6`
or a major (usually breaking) version change
i.e `openssl 1.1.x` to `3.x`

"[Semantic versioning](https://semver.org/)", applied more or less faithfully - is a common versioning scheme.
But some projects such as GNU Parallel instead prefer to name releases by date,
for example the latest version of parallel right now is `20230822` and a new "stable" version usually releases on the 22nd of each month.

What constitutes a "stable" release varies from project to project,
In lieu of clear versions, specific git commits or tags are an acceptable substitute on a case by case basis.

The [checklist for submitting new packages](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#submitting-new-packages-checklist) goes a bit more into it regarding version naming rules in non-standard cases.

**Comment by sylirre** [comment, marked as answer]
Termux does not implement versioning like Debian does. We don't perform division between stable and testing distributions. Termux is rolling release.

We are not that strict about determining whether software is stable and can be packaged. This boils down to such principles (they ARE NOT rules and not mentioned in docs):

* Use the latest available version which works. It is preferred that version is a production grade release (not marked explicitly as alpha/beta/development). If such one is not available or is too much outdated, we would accept alpha/beta versions or even specific Git commits.
* If package is a dependency of others, the new version must be backwards compatible. Thus if library in the next major version made incompatible changes, we can deny upgrading until all reverse dependent packages made necessary changes to be compatible with new release of this library.
* Critical packages such as `apt`, `bash`, `coreutils`, `dpkg`, `tar` should go through testing before upgrade is being pushed. This is not necessary for other packages.

You may notice that many packages are upgraded automatically by GitHub Actions bot without intervention from our side. Termux has no resources for thorough QA.
